pf4j/src/main/java/org/pf4j/SingletonExtensionFactory.java

```diff
@@ -25,12 +25,13 @@
  * Optional, you can specify the extension classes for which you want singletons.
  *
  * @author Decebal Suiu
+ * @author Ajith Kumar
  */
 public class SingletonExtensionFactory extends DefaultExtensionFactory {
 
     private final List<String> extensionClassNames;
 
-    private Map<String, Object> cache;
+    private Map<ClassLoader, Map<String, Object>> cache;
```

Comment:
We can consider using a `WeakHashMap` as the backing implementation to prevent classloader instances from being skipped during GC cycles when no other references to it exists and thus preventing leaks.

Reply:
Replace `HashMap` with a `WeakHashMap`.

Reply:
Changed to `WeakHashMap`.